server/src/main/java/org/elasticsearch/action/admin/indices/get/GetIndexResponse.java

```diff
@@ -355,6 +387,10 @@ private static IndexEntry parseIndexEntry(XContentParser parser) throws IOExcept
     }
 
     public static GetIndexResponse fromXContent(XContentParser parser) throws IOException {
+        return fromXContent(parser, false);
+    }
+
+    public static GetIndexResponse fromXContent(XContentParser parser, boolean legacyWithTypes) throws IOException {
```

Comment:
To check I understand, are we only retaining this method so that we can test the serialization of `GetIndexResponse`? It looks like we do not support `include_type_name = true` in the get indices HLRC call.

Reply:
Yes, at some point I was planning on adding an alternative get-method (something like getWithTypes) to the IndicesClient, but now I think we don't need it unless we want ppl. to be able to get back their legacy type names from 6.x indices from the Java-Objects returned by the HLRC. If we are fine with this not being possible anymore I think we don't need this.

Reply:
I agree that this is not so critical, as users should stop using the type name to encode information. So we could keep things simple and not add another method `getWithTypes`.